### PR TITLE
Define django model for users

### PR DESCRIPTION
**Pull request by gogolmariana**
Defined Django model for Users. Created migrations.

### CODE REVIEW

File: forum/users/models.py [hunk 1]
```diff
@@ -1,3 +1,26 @@
+from django.core.validators import EmailValidator
 from django.db import models
+import uuid
 
 # Create your models here.
+class Role(models.Model):
+    role_id = models.UUIDField(primary_key=True, default=uuid.uuid4)
```
Comment (mehalyna) [inline forum/users/models.py:7]: Consider adding `editable=False` to ensure that users don’t attempt to change the UUID.
```
role_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
```

File: forum/users/models.py [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+    role_name = models.CharField(max_length=50, null=False, unique=True)
+
+class User(models.Model):
+    user_id = models.UUIDField(primary_key=True, default=uuid.uuid4)
```
Comment (mehalyna) [inline forum/users/models.py:11]: Consider adding `editable=False` to ensure that users don’t attempt to change the UUID.
_example in line's 7 comment_

File: forum/users/models.py [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+    user_name = models.CharField(max_length=100, null=False, unique=True)
+    first_name = models.CharField(max_length=100, null=False)
+    last_name = models.CharField(max_length=100, null=False)
+    email = models.EmailField(unique=True, null=False, max_length=50, validators=[EmailValidator()])
```
Comment (mehalyna) [inline forum/users/models.py:15]: Using `EmailField` with `EmailValidator()` is redundant. The `EmailField` already includes basic email validation, so adding `EmailValidator()` is unnecessary.
Also, consider increasing the `max_length` for the email field. Standard practice is to use a maximum of 254 characters for email fields, as defined by `RFC 5321`.
https://datatracker.ietf.org/doc/html/rfc5321

File: forum/users/models.py [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+    password = models.CharField(max_length=50)
```
Comment (mehalyna) [inline forum/users/models.py:16]: `max_length=50` may not be sufficient if you're storing hashed passwords. Password hashes can easily exceed 50 characters, especially when using secure algorithms

File: forum/users/models.py [hunk 5]
```diff
@@ -1,3 +1,26 @@
...
+    user_phone = models.CharField(max_length=15)
```
Comment (mehalyna) [inline forum/users/models.py:17]: There’s no validation in place. You should consider adding a custom validator 
_Custom validation example_
```
from django.core.validators import RegexValidator

phone_validator = RegexValidator(
    regex=r'^\+?1?\d{9,15}$',
    message="Phone number must be entered in the format: '+999999999'. Up to 15 digits allowed."
)

user_phone = models.CharField(validators=[phone_validator], max_length=15)
```
Alternatively, you can use Django’s `PhoneNumberField` (from the `django-phonenumber-field` package) to validate phone numbers.

File: forum/users/models.py
Comment (mehalyna) [inline forum/users/models.py:1]: You can add `verbose_name` and `verbose_name_plural` to make the model more readable in the Django admin.
_Example to the class `Role`:_
```
    class Meta:
        verbose_name = "Role"
        verbose_name_plural = "Roles"
```
The same - for the class `User`

File: forum/users/models.py [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+    roles = models.ManyToManyField(Role)
```
Comment (mehalyna) [inline forum/users/models.py:18]: Consider adding `related_name` to make reverse lookups more intuitive:
```
roles = models.ManyToManyField(Role, related_name="users")
```

**Review by mehalyna — CHANGES_REQUESTED** [review]

File: forum/users/models.py (hunk 4 above)
Comment (gogolmariana) [reply forum/users/models.py:16]: Set max_length=255 for password field.

File: forum/users/models.py (hunk 3 above)
Comment (gogolmariana) [reply forum/users/models.py:15]: Removed validator from email field and set max_length=255.

File: forum/users/models.py (hunk 2 above)
Comment (gogolmariana) [reply forum/users/models.py:11]: Added editable=False for user_id field.

File: forum/users/models.py (hunk 1 above)
Comment (gogolmariana) [reply forum/users/models.py:7]: Added editable=False for user_id field.

File: forum/users/models.py (hunk 6 above)
Comment (gogolmariana) [reply forum/users/models.py:18]: Added related_name="users" to roles field.

File: forum/users/models.py
Comment (gogolmariana) [reply forum/users/models.py:1]: Сhanges made for better readability.

File: forum/users/models.py (hunk 5 above)
Comment (gogolmariana) [reply forum/users/models.py]: Used **django-phonenumber-field** validator

**Review by mehalyna — APPROVED** [review]

**Review by Chelakhovl — APPROVED** [review]

**Review by k0drin — APPROVED** [review]

**Pull request merged**